pkg/ddl/ddl_worker.go

```diff
@@ -1008,13 +1008,18 @@ func (w *worker) HandleDDLJobTable(d *ddlCtx, job *model.Job) (int64, error) {
 	return schemaVer, nil
 }
 
-func (w *worker) checkOwnerBeforeCommit() error {
+func (w *worker) checkBeforeCommit() error {
 	if !w.ddlCtx.isOwner() && w.tp != localWorker {
 		// Since this TiDB instance is not a DDL owner anymore,
 		// it should not commit any transaction.
 		w.sess.Rollback()
 		return dbterror.ErrNotOwner
 	}
+
+	if err := w.ctx.Err(); err != nil {
+		// The worker context is canceled, it should not commit any transaction.
+		return err
```

Comment:
this mitigates the problem and the better way is pass `w.ctx` to the function where it's really used. Maybe `w.sess.RollbackWithCtx()`, `w.sess.RollbackWithCtx()` and elsewhere.

Reply:
We do have the context-attached version. However, these contexts are neither used nor checked correctly.

```
s.StmtRollback(context.Background(), false)
s.RollbackTxn(context.Background())
```